R/shiny.R

```diff
@@ -1876,6 +1876,22 @@ ShinySession <- R6Class(
     # Provides a mechanism for handling direct HTTP requests that are posted
     # to the session (rather than going through the websocket)
     handleRequest = function(req) {
+      if (!is.null(self$user)) {
+        if (is.null(req$HTTP_SHINY_SERVER_CREDENTIALS)) {
+          # Session owner is logged in, but this requester is not
+          return(NULL)
+        }
+
+        creds <- NULL
+        try({creds <- safeFromJSON(req$HTTP_SHINY_SERVER_CREDENTIALS)},
+          silent = TRUE
+        )
+        if (!identical(self$user, creds$user)) {
```

Comment:
This could throw an error if `creds` is not a list or `NULL`. For example, it could be an atomic vector.

```R
x <- shiny:::safeFromJSON("[1,2,3]")
x$user
#> Error in x$user : $ operator is invalid for atomic vectors
```

So maybe this part should also be inside of the `try`.